Source/Core/VideoBackends/D3D/D3DBase.h

```diff
@@ -43,6 +43,9 @@ std::vector<u32> GetAAModes(u32 adapter_index);
 // Checks for support of the given texture format.
 bool SupportsTextureFormat(DXGI_FORMAT format);
 
+// Checks for logic op support.
+bool SupportsLogicOp(u32 adapter_index);
```

Comment:
I don't think we need to create a temporary device for testing logic op support. You can populate it at the same time as the other optional features: https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/VideoBackends/D3D/main.cpp#L94

While we probably should be checking against the actual device for things like bounding box as well, the device creation should be part of `FillBackendInfo()`, not `SupportsLogicOp()`.

Reply:
Good call! This change predates the commit with a warning message, back then I assumed I will use `g_Config.backend_info.bSupportsLogicOp` for this warning message but this was deemed impossible, so I am now calling `SupportsLogicOp` directly when returning the warning message. This can be moved to inside that code branch, then. Will do.